source/draw/RasterYuv420Semiplanar.ooc

```diff
@@ -35,12 +61,19 @@ RasterYuv420Semiplanar: class extends RasterYuvSemiplanar {
 	init: func ~allocate (size: IntVector2D) { this init(size, size x) }
 	init: func ~fromThis (original: This) {
 		(yImage, uvImage) := This _allocate(original size, original stride, original stride * original size y)
-		super(original, yImage, uvImage)
+		this init(original, yImage, uvImage)
 	}
 	init: func ~fromByteBuffer (buffer: ByteBuffer, size: IntVector2D, stride: UInt, uvOffset: UInt) {
 		(yImage, uvImage) := This _createSubimages(buffer, size, stride, uvOffset)
 		this init(yImage, uvImage)
 	}
+	free: override func {
+		(this y, this uv) referenceCount decrease()
+		super()
+	}
+	distance: override func (other: Image) -> Float {
+		(this y distance((other as This) y) + this uv distance((other as This) uv)) / 2.0f
```

Comment:
This was simplified and should be a lot faster by not reading each UV pixel 4 times in Y resolution.